kornia/geometry/camera/stereo.py

```diff
@@ -0,0 +1,287 @@
+            )
+
+        # Ensure all intrinsics parameters (fx, fy, cx, cy) are the same in both cameras.
+        if not torch.all(torch.eq(rectified_left_camera[..., :, :3], rectified_right_camera[..., :, :3])):
+            raise StereoException(
+                f"Expected 'left_rectified_camera' and 'rectified_right_camera' to have"
+                f"same parameters except for the last column."
+                f"Got {rectified_left_camera[..., :, :3]} and {rectified_right_camera[..., :, :3]}."
+            )
+
+        # Ensure that tx * fx is negative and exists.
+        tx_fx = rectified_right_camera[..., 0, 3]
+        if torch.all(torch.gt(tx_fx, 0)):
+            raise StereoException(f"Expected :math:`T_x * f_x` to be negative." f"Got {tx_fx}.")
+
+    @property
+    def batch_size(self) -> int:
+        r"""Returns the batch size of the storage.
+
+        Returns:
+            int: scalar with the batch size
+        """
+        return self.rectified_left_camera.shape[0]
+
+    @property
+    def fx(self) -> torch.Tensor:
+        r"""Returns the focal length in the x-direction. Note that the focal lengths of the rectified left and right
+        camera are assumed to be equal.
+
+        Returns:
+            torch.Tensor: tensor of shape :math:`(B)`
+        """
+        return self.rectified_left_camera[..., 0, 0]
+
+    @property
+    def fy(self) -> torch.Tensor:
+        r"""Returns the focal length in the y-direction. Note that the focal lengths of the rectified left and right
+        camera are assumed to be equal.
+
+        Returns:
+            torch.Tensor: tensor of shape :math:`(B)`
+        """
+        return self.rectified_left_camera[..., 1, 1]
+
+    @property
+    def cx_left(self) -> torch.Tensor:
+        r"""Returns the x-coordinate of the principal point for the left camera.
+
+        Returns:
+            torch.Tensor: tensor of shape :math:`(B)`
+        """
+        return self.rectified_left_camera[..., 0, 2]
+
+    @property
+    def cx_right(self) -> torch.Tensor:
+        r"""Returns the x-coordinate of the principal point for the right camera.
+
+        Returns:
+            torch.Tensor: tensor of shape :math:`(B)`
+        """
+        return self.rectified_right_camera[..., 0, 2]
+
+    @property
+    def cy(self) -> torch.Tensor:
+        r"""Returns the y-coordinate of the principal point. Note that the y-coordinate of the principal points
+        is assumed to be equal for the left and right camera.
+
+        Returns:
+            torch.Tensor: tensor of shape :math:`(B)`
+        """
+        return self.rectified_left_camera[..., 1, 2]
+
+    @property
+    def tx(self) -> torch.Tensor:
+        """
+        The horizontal baseline between the two cameras.
+
+        Returns:
+            torch.tensor: Tensor of shape :math:`(B)`
+
+        """
+        return -self.rectified_right_camera[..., 0, 3] / self.fx
+
+    @property
+    def Q(self) -> torch.Tensor:
```

Comment:
is there any way or makes sense to cache this ?

Reply:
Yes, see update.